watertap/unit_models/tests/test_anaerobic_digester.py

```diff
@@ -238,3 +257,365 @@ def configure(self):
+        m = ConcreteModel()
+        m.fs = FlowsheetBlock(dynamic=False)
+
+        m.fs.props = ADM1ParameterBlock()
+        m.fs.props_vap = ADM1_vaporParameterBlock()
+        m.fs.rxn_props = ADM1ReactionParameterBlock(property_package=m.fs.props)
+
+        m.fs.unit = AD(
+            liquid_property_package=m.fs.props,
+            vapor_property_package=m.fs.props_vap,
+            reaction_package=m.fs.rxn_props,
+            has_heat_transfer=True,
+            has_pressure_change=False,
+        )
+
+        # Set the operating conditions
+        m.fs.unit.inlet.flow_vol.fix(170 / 24 / 3600)
+        m.fs.unit.inlet.temperature.fix(308.15)
+        m.fs.unit.inlet.pressure.fix(101325)
+
+        m.fs.unit.inlet.conc_mass_comp[0, "S_su"].fix(0.01)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_aa"].fix(0.001)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_fa"].fix(0.001)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_va"].fix(0.001)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_bu"].fix(0.001)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_pro"].fix(0.001)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_ac"].fix(0.001)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_h2"].fix(1e-8)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_ch4"].fix(1e-5)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_IC"].fix(0.48)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_IN"].fix(0.14)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_I"].fix(0.02)
+
+        m.fs.unit.inlet.conc_mass_comp[0, "X_c"].fix(2)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_ch"].fix(5)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_pr"].fix(20)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_li"].fix(5)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_su"].fix(0.0)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_aa"].fix(0.010)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_fa"].fix(0.010)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_c4"].fix(0.010)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_pro"].fix(0.010)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_ac"].fix(0.010)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_h2"].fix(0.010)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_I"].fix(25)
+
+        m.fs.unit.inlet.cations[0].fix(0.04)
+        m.fs.unit.inlet.anions[0].fix(0.02)
+
+        m.fs.unit.volume_liquid.fix(3400)
+        m.fs.unit.volume_vapor.fix(300)
+        m.fs.unit.liquid_outlet.temperature.fix(308.15)
+
+        sb = ScalerBase()
+
+        # Apply scaling to unscaled variables
+        for var in m.fs.component_data_objects(Var, descend_into=True):
+            if "conc_mass_comp" in var.name:
+                sb.set_variable_scaling_factor(var, 1e1)
+            if "conc_mol" in var.name:
+                sb.set_variable_scaling_factor(var, 1e2)
+            if "reaction_rate" in var.name:
+                sb.set_variable_scaling_factor(var, 1e6)
+
+            sb.set_variable_scaling_factor(m.fs.unit.hydraulic_retention_time[0], 1e-6)
```

Comment:
Why do this? Is this just the "new" equivalent to what we were doing with the old way of scaling (i.e., create a ScalerBase and set variable scaling as desired to certain variables before applying the unit's Scaler)?

Reply:
Pretty much, although this should probably be applied after the unit's Scaler since we want the latter to take priority. The way things are currently set up, the unit scaler's config option for overwrite is set to False, so it will not overwrite any pre-existing scaling factors.